### PR TITLE
add user-defined address families (src/addr_families.c etc.)

### PR DESCRIPTION
**Pull request by douzzer**
includes new coupled-table handling (`wolfsentry_coupled_table_clone()` etc.).  moves `wolfsentry_addr_family_pton()` and `wolfsentry_addr_family_ntop()` to src/addr_families.c.  also, adds code to properly allocate new object IDs when cloning objects.

### CODE REVIEW

File: src/addr_families.c [hunk 1]
```diff
@@ -0,0 +1,832 @@
...
+        } else if (! WOLFSENTRY_ERROR_CODE_IS(ret, ITEM_NOT_FOUND)) {
+            if (errcode != NULL)
+                *errcode = ret;
+            return WOLFSENTRY_AF_UNSPEC;
+        }
+    }
+
+    if (errcode)
+        *errcode = WOLFSENTRY_ERROR_ENCODE(OK);
+    if (strcaseeq(family_name, "UNIX", (size_t)family_name_len))
```
Comment (LinuxJedi) [inline src/addr_families.c:493]: Would this section be better as a lookup table?

File: src/addr_families.c [hunk 2]
```diff
@@ -0,0 +1,832 @@
...
+}
+
+static const char *wolfsentry_addr_family_ntop_1(
+    wolfsentry_addr_family_t family,
+    wolfsentry_errcode_t *errcode)
+{
+    if (errcode != NULL)
+        *errcode = WOLFSENTRY_ERROR_ENCODE(OK);
+    switch(family) {
+    case WOLFSENTRY_AF_UNSPEC:
```
Comment (LinuxJedi) [inline src/addr_families.c:642]: Likewise I think this might work as a LUT?

**Review by LinuxJedi — APPROVED** [review]
Approved with a couple of comments.

File: src/addr_families.c (hunk 1 above)
Comment (douzzer) [reply src/addr_families.c:493]: doubtless better as a lookup table.  let's leave it lame for now (avoid premature optimization) and promise we'll fix it later.

the biggest yet-unseized optimization opportunity is red-black tables for the main tables.  it's on the roadmap.

File: src/addr_families.c (hunk 2 above)
Comment (douzzer) [reply src/addr_families.c:642]: likewise agree.

**Pull request merged**